src/components/resource/edit-managed-title/resource-edit-managed-title.js

```diff
@@ -33,60 +32,36 @@ class ResourceEditManagedTitle extends Component {
     initialValues: PropTypes.object.isRequired,
     intl: intlShape.isRequired,
     model: PropTypes.object.isRequired,
+    onCancel: PropTypes.func.isRequired,
     onSubmit: PropTypes.func.isRequired,
     pristine: PropTypes.bool,
     proxyTypes: PropTypes.object.isRequired
   };
 
-  static contextTypes = {
-    router: PropTypes.shape({
-      history: PropTypes.shape({
-        push: PropTypes.func.isRequired
-      }).isRequired
-    }).isRequired
-  };
-
   state = {
     managedResourceSelected: this.props.initialValues.isSelected,
     showSelectionModal: false,
     allowFormToSubmit: false,
-    formValues: {}
+    formValues: {},
+    initialValues: this.props.initialValues
   }
 
-  componentWillReceiveProps(nextProps) { // eslint-disable-line react/no-deprecated
-    let wasPending = this.props.model.update.isPending && !nextProps.model.update.isPending;
-    let needsUpdate = !isEqual(this.props.model, nextProps.model);
-
-    let wasUnSelected = this.props.model.isSelected && !nextProps.model.isSelected;
-    let isCurrentlySelected = this.props.model.isSelected && nextProps.model.isSelected;
-
-    if (nextProps.initialValues.isSelected !== this.state.managedResourceSelected) {
-      this.setState({
-        managedResourceSelected: nextProps.initialValues.isSelected,
-      });
-    }
-
+  static getDerivedStateFromProps(nextProps, prevState) {
     if (nextProps.model.update.errors.length) {
-      this.setState({
-        showSelectionModal: false
-      });
+      return { showSelectionModal: false };
     }
 
-    if (wasUnSelected || isCurrentlySelected) {
-      if (wasPending && needsUpdate) {
-        this.context.router.history.push(
-          `/eholdings/resources/${this.props.model.id}`,
-          { eholdings: true, isFreshlySaved: true }
-        );
-      }
+    if (nextProps.initialValues.isSelected !== prevState.initialValues.isSelected) {
+      return {
+        ...prevState,
```

Comment:
The return from this method _updates_ state, so we don't need to include (or return below) the previous state.

Reply:
Removed all instances of `getDerivedStateFromProps` returning `prevState`.